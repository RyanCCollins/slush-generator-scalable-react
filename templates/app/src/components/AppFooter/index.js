import React from 'react';
import cssModules from 'react-css-modules';
import Footer from 'grommet-udacity/components/Footer';
import Box from 'grommet-udacity/components/Box';
import Heading from 'grommet-udacity/components/Heading';
import SocialShare from 'grommet-udacity/components/SocialShare';
import Anchor from 'grommet-udacity/components/Anchor';
import styles from './index.module.scss';

const AppFooter = () => (
  <Footer pad="large" colorIndex="light-2">
    <Box
      direction="column"
      align="center"
      pad="none"
      responsive
      className={styles.flexOne}
    >
      <Heading tag="h3">
        By{' '}
        <a
          className={styles.footerAnchor}
          href="http://www.ryancollins.io"
        >
          Ryan Collins
        </a>
      </Heading>
      <Heading tag="h5">
        This app is licensed under the{' '}
        <a
          className={styles.footerAnchor}
          href="https://github.com/RyanCCollins/ryancollinsio/blob/master/LICENSE"
        >
          MIT License.
        </a>
        {' '}Take a peak at the{' '}
        <br />
        <Anchor
          className={styles.footerAnchor}
          href="https://github.com/RyanCCollins/ryancollinsio"
        >
          source code.
        </Anchor>
      </Heading>
      <nav
        aria-hidden
        className={
          'grommetux-box grommetux-box--direction-row ' +
          'grommetux-box--responsive grommetux-box--pad-none ' +
          'grommetux-menu grommetux-menu--row grommetux-menu--inline'
        }
      >
        <SocialShare
          a11yTitle="Go to Facebook to Share this website"
          type="facebook"
          link="http://www.ryancollins.io"
          text="RyanCollins.io"
        />
        <SocialShare
          a11yTitle="Go to Twitter to Share this website"
          type="twitter"
          link="http://www.ryancollins.io"
          text="RyanCollins.io"
        />
        <SocialShare
          a11yTitle="Go to Linkedin to Share this website"
          type="linkedin"
          link="http://www.ryancollins.io"
          title="Restaurant Reviewer"
          text="RyanCollins.io"
        />
      </nav>
    </Box>
  </Footer>
);

export default cssModules(AppFooter, styles);
