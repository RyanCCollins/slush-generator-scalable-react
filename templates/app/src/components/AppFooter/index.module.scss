.appFooter {
  background: #fbfbfb;
  padding: 50px;
  title {
    color: black !important;
  }
}

.flexOne {
  flex: 1;
}
