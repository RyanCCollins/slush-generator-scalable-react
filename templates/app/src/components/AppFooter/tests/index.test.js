/* global expect b:true */
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import AppFooter from '../index';

describe('<AppFooter />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <AppFooter />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
