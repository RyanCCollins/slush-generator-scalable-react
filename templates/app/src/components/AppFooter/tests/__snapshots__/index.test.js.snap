exports[`<AppFooter /> should render with default props 1`] = `
<Footer
  colorIndex="light-2"
  direction="row"
  pad="large"
  responsive={false}>
  <Box
    align="center"
    announce={false}
    direction="column"
    focusable={true}
    pad="none"
    responsive={true}
    tag="div">
    <Heading
      tag="h3">
      By
       
      <a
        href="http://www.ryancollins.io">
        Ryan Collins
      </a>
    </Heading>
    <Heading
      tag="h5">
      This app is licensed under the
       
      <a
        href="https://github.com/RyanCCollins/ryancollinsio/blob/master/LICENSE">
        MIT License.
      </a>
       
      Take a peak at the
       
      <br />
      <Anchor
        href="https://github.com/RyanCCollins/ryancollinsio"
        tag="a">
        source code.
      </Anchor>
    </Heading>
    <nav
      aria-hidden={true}
      className="grommetux-box grommetux-box--direction-row grommetux-box--responsive grommetux-box--pad-none grommetux-menu grommetux-menu--row grommetux-menu--inline">
      <SocialShare
        a11yTitle="Go to Facebook to Share this website"
        link="http://www.ryancollins.io"
        text="RyanCollins.io"
        title=""
        type="facebook" />
      <SocialShare
        a11yTitle="Go to Twitter to Share this website"
        link="http://www.ryancollins.io"
        text="RyanCollins.io"
        title=""
        type="twitter" />
      <SocialShare
        a11yTitle="Go to Linkedin to Share this website"
        link="http://www.ryancollins.io"
        text="RyanCollins.io"
        title="Restaurant Reviewer"
        type="linkedin" />
    </nav>
  </Box>
</Footer>
`;
